packages/react-components/react-nav-preview/src/components/NavContext.ts

```diff
@@ -0,0 +1,30 @@
+import { createContext, useContextSelector } from '@fluentui/react-context-selector';
+import type { Context, ContextSelector } from '@fluentui/react-context-selector';
+import { NavContextValue } from './NavContext.types';
+
+const navContextDefaultValue: NavContextValue = {
+  reserveSelectedNavGroupSpace: true,
+  selectedValue: undefined,
+  onRegister: () => {
+    /* noop */
+  },
+  onUnregister: () => {
+    /* noop */
+  },
+  onSelect: () => {
+    /* noop */
+  },
+  getRegisteredNavGroups: () => {
+    return {
+      registeredNavGroups: {},
+    };
+  },
+};
+
+export const NavContext: Context<NavContextValue> = createContext<NavContextValue | undefined>(
+  undefined,
+) as Context<NavContextValue>;
+
+export const NavProvider = NavContext.Provider;
+export const useNavContext_unstable = <T>(selector: ContextSelector<NavContextValue, T>): T =>
+  useContextSelector(NavContext, (ctx = navContextDefaultValue) => selector(ctx));
```

Comment:
ContextSelector is a relatively big overhead in terms of perf and bundle size. It's only worthwhile if:
* The context has some values that change frequently, and some that rarely change.
* Most components that use the context don't depend on the frequently-changing values.

You may want to consider using a vanilla context here. See this PR that refactored RadioGroup to remove ContextSelector for an example (also note the huge bundle size win from the change):
* https://github.com/microsoft/fluentui/pull/27476

Reply:
Make sure to memoize the context value in` useNavContextValues_unstable` if you make this change.

Reply:
Okay - got that rejiggered drafting off the radio group example. Have a conversation with Geoff going to understand why he used the selector in the first place.

Reply:
Switched to vanilla react context. Let me know if this doesn't look right.